engine/swarm/secrets.md

```diff
@@ -0,0 +1,339 @@
+
+    ```bash
+    $ docker service rm redis
+
+    $ docker sercret rm secret_data.txt
+    ```
+
+## Example: Use secrets with a service
+
+In this example, you create a MySQL service with a custom username and password.
+Next, you add these credentials as secrets. Next, you create a WordPress
+service which uses these credentials to connect to MySQL. Finally, you update
+the MySQL password and update the secret, so that the WordPress service can
+still connect to MySQL.
+
+**Note**: This example uses a single-Engine swarm for simplicity.
+
+1.  On a new Docker Engine install, initialize a new swarm.
+    ```bash
+    $ docker swarm init
+    ```
+
+    This Engine is now the manager of a new swarm. If you join additional
+    managers or workers to this swarm, they will receive the key used to encrypt
+    and decrypt the secrets over a TLS-encrypted connection.
+
+2.  Create a new overlay network `wp_network` which the MySQL and WordPress
+    services will use to communicate.
+    ```bash
+    $ docker network create -d overlay wp_network
+    ```
+
+3.  Generate a random alphanumeric password for MySQL and store it as a Docker
+    secret with the name `mysql_password` using the `docker secret create`
+    command. To make the password shorter or longer, adjust the last argument of
+    the `openssl` command. This is just one way to create a relatively random
+    password. You can use another command to generate the password if you choose.
+    Notice that the password is never exposed on the command line or saved to a
+    plain-text file.
+
+    >**Note**: After you create a secret, you cannot update it. You can only
+    remove and re-create it, and you cannot remove a secret that a service is
+    using. However, you can grant or revoke a running service's access to
+    secrets using `service update`. If you need the ability to update a secret,
+    consider adding a version component to the secret name, so that you can later
+    add a new version, update the service to use it, then remove the old version.
+
+    ```bash
+    $ openssl rand -base64 20 |docker secret create mysql_password
+      lwxgks2rojdbvaui5kay1suvr
+    ```
+
+    The value returned is not the password, but the ID of the secret.
+    List the secrets managed by Docker using `docker secret ls`:
+
+    ```bash
+    $ docker secret ls
+    ID                          NAME                CREATED             UPDATED
+    lwxgks2rojdbvaui5kay1suvr   mysql_password      41 seconds ago      41 seconds ago
+    ```
+    The secret is stored in the encrypted RAFT logs for the swarm.
+
+4.  Create the MySQL service. The `mysql` image now supports additional
+    environment variables which allows the container to read credentials and other
+    important settings from files within the container. This example uses the new
+    `MYSQL_ROOT_PASSWORD_FILE` variable and sets it to the file path within the
+    container which contains the `mysql_password` secret.
```

Comment:
Should there be a link to the readme? "For an overview of all environment variables available for configuring the mysql image, refer to the image's documentation on Docker Hub https://hub.docker.com/_/mysql/

(or Docker Store; https://store.docker.com/images/3083290a-203f-4c04-b2de-cc057959d2c9?tab=description)